cuML/test/quasi_newton.cu

```diff
@@ -45,17 +49,17 @@ const double QuasiNewtonTest::X[QuasiNewtonTest::N][QuasiNewtonTest::D] = {
 
 template <typename T, class Comp>
 ::testing::AssertionResult
-checkParamsEqual(const T *host_weights, const T *host_bias, const T *w,
-                 const int C, const int D, const bool fit_intercept,
-                 Comp &comp) {
+checkParamsEqual(const cumlHandle_impl &handle, const T *host_weights,
+                 const T *host_bias, const T *w, const int C, const int D,
+                 const bool fit_intercept, Comp &comp) {
   std::vector<T> w_ref_cm(C * D);
   int idx = 0;
   for (int d = 0; d < D; d++)
     for (int c = 0; c < C; c++) {
       w_ref_cm[idx++] = host_weights[c * D + d];
     }
 
-  SimpleVecOwning<T> w_ref(C * (D + fit_intercept));
+  SimpleVecOwning<T> w_ref(handle, C * (D + fit_intercept), handle.getStream());
   updateDevice(w_ref.data, &w_ref_cm[0], C * D);
```

Comment:
Just pointing out that `updateDevice` possible relies on default stream semantics, i.e. that it implicitly syncs with `handle.getStream()`. This will break if the stream passed into `handle` is created with the non blocking flag. This also applies to other parts of the test.